magento/magento2
# CSS file is deferred when CSS element has defer attribute in default_head_blocks.xml.

### Description (*)
While I searching for ways to defer non-critical CSS files, I found that magento2 has a `defer` attribute in the [head.xsd](https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/View/Layout/etc/head.xsd#L11C27-L11C27) file. This means I can write the following code in the `default_head_blocks.xml` without any errors.

``` xml
<css src="css/intl-tel-input.css" defer="true" />
```

But I realized that this `defer` attribute has no effect on the CSS element. Because HTML [link element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attributes) doesn't have `defer` attribute. It only works with the `<script>` element. This pull request aims to provide defer feature for CSS files with [defer non-critical render-blocking CSS](https://web.dev/defer-non-critical-css/) hacky method.

### Related Pull Requests
1. Fixes magento/magento2#32919

### Fixed Issues (if relevant)
1. Fixes magento/magento2#37657

### Manual testing scenarios (*)
1. Create/Open `default_head_blocks.xml` file and add a CSS element like below;
``` xml
<css src="css/intl-tel-input.css" defer="true" />
```
2. Flush cache (`bin/magento cache:flush`)
3. Open your Magento2 site in the browser and press `ctrl` + `u` combination.
4. See that the CSS file is defined as deferred like below.
``` html
<link rel="preload" type="text/css" as="style" onload="this.onload=null;this.rel=&#039;stylesheet&#039;" href="https://magento.test/static/version1690544175/frontend/ExampleVendor/base/tr_TR/css/intl-tel-input.css" />
<noscript><link rel="stylesheet" href="https://magento.test/static/version1690544175/frontend/ExampleVendor/base/tr_TR/css/intl-tel-input.css"></noscript>
```
5. Disable Javascript in site settings on the browser. Be sure that the css file loaded properly.



### Questions or comments
I am not entirely sure that these changes are correct and I don't know if it will affect other code. I tested it on my browser and it works fine. Please review my code and let me know if I made any mistakes.

### Contribution checklist (*)
